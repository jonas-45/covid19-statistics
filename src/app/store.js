import { configureStore } from '@reduxjs/toolkit';
import leaguesSliceReducer from '../redux/leagues/leaguesSlice';

const store = configureStore({
  reducer: {
    leagues: leaguesSliceReducer,
  },
});

export default store;
