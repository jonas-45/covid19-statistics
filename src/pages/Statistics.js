import React from 'react';
import { Card } from 'react-bootstrap';
import { useSelector } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleArrowRight } from '@fortawesome/free-solid-svg-icons';

const Statistics = () => {
  const { covidStats } = useSelector((state) => state.covid);
  if (covidStats.length < 1) {
    return (
      <div>
        <p>Loading data, please wait...</p>
      </div>
    );
  }
  return (
    <div className="d-flex flex-wrap justify-content-center gap-3 p-3">
      {covidStats.map((info) => (
        <Card style={{ width: '18rem' }} key={info.id} className="bg-light p-2">
          <Card.Img variant="top" src={info.flag} style={{ width: '100%', height: '50%' }} />
          <Card.Body>
            <Card.Title>{info.country }</Card.Title>
            <div className="d-flex justify-content-between align-items-center">
              <Card.Subtitle>
                Deaths:
                {' '}
                {info.deaths}
              </Card.Subtitle>
              <Card.Subtitle><FontAwesomeIcon icon={faCircleArrowRight} size="2x" /></Card.Subtitle>
            </div>
            <Card.Text className='mt-3'>
              Polulation:
              {' '}
              {info.population}
            </Card.Text>
          </Card.Body>
        </Card>
      ))}
    </div>
  );
};

export default Statistics;
