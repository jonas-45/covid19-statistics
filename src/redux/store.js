import { configureStore } from '@reduxjs/toolkit';
import covidSliceReducer from './covid/covidSlice';

const store = configureStore({
  reducer: {
    covid: covidSliceReducer,
  },
});

export default store;
