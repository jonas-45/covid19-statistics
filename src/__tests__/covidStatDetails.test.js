import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Statistics from '../pages/Statistics';
import store from '../redux/store';

describe('Test for countryDetails', () => {
  test('should render', () => {
    const country = {
      id: 13,
      death: '1933',
      critical: '0',
      country: 'Angola',
      active: '23',
      cases: '105298',
      recovered: '103342',
      tests: '1499795',
      today: '123',
    };
    const tree = renderer.create(
      <Provider store={store}>
        <Statistics countries={country} key={24} />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
