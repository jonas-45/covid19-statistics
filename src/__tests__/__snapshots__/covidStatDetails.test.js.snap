// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for countryDetails should render 1`] = `
<div
  className="text-center mt-5 pt-5"
>
  <p>
    Loading data, please wait...
  </p>
</div>
`;
